[14.0] [IMP] account_statement_import_online[_ponto] improvements.

NL66278 opened the pull request:
1. Find partner if there already is a bank account with the right IBAN;
2. Store (and display) raw import data to help in problem determination;
3. Make it easy to extend the parsing of import data;
4. Apply some clean coding principles.

NL66278 committed:
[IMP] account_statement_import_online[_ponto] improvements.

1. Find partner if ther already is a bank account with the right IBAN;
2. Store (and display) raw import data to help in problem determination;
3. Make it easy to extend the parsing of import data;
4. Apply some clean coding principles.

pedrobaeza commented on account_statement_import_online/models/account_bank_statement_line.py:
```diff
@@ -0,0 +1,18 @@
+
+    _inherit = "account.bank.statement.line"
+
+    raw_data = fields.Text(
```
```suggestion
    online_raw_data = fields.Text(
```
?

NL66278 replied:
I will change the name

pedrobaeza commented on account_statement_import_online/models/online_bank_statement_provider.py:
```diff
@@ -365,6 +363,14 @@ def _sanitize_bank_account_number(self, bank_account_number):
 
+    def _update_partner_from_account_number(self, line_values):
+        partner_bank = self.env["res.partner.bank"].search(
+            [("acc_number", "=", line_values["account_number"])], limit=1
```
No `company_id` restriction?

NL66278 replied:
I took this from the not online account_statement_import. res.partner.bank statements with company_id filled are own accounts, not partner accounts. Maybe I should add that company_id should be False.

pedrobaeza replied:
Can you have the same bank account number in several companies? If so, then you should check by company_id = company of the statement or company_id = False.

NL66278 replied:
O wait, that was in in old version of Odoo.

NL66278 replied:
Still, I do not see a need to check for company. You can have a partner without a company_id, meaning valid everywhere, and then when adding a bank_account, the res.partner.bank record gets the company_id of the logged in user. I think that is illogical, and unneeded, as it is still not a required field.

pedrobaeza replied:
I insist: the problem is that you have the same bank account number in several companies, and with this, you select the incorrect bank account belonging to other company.

pedrobaeza commented on account_statement_import_online_ponto/models/online_bank_statement_provider_ponto.py:
```diff
@@ -156,34 +156,32 @@ def _ponto_get_transaction(self, account_id, date_since, date_until):
                     _("Error during get transaction.\n\n%s \n\n %s")
                     % (response.status_code, response.text)
                 )
+            if params.get("before"):
```
Why changing this order? Please try to minimize diff

NL66278 replied:
I thought it was cleaner, saving a lot of indented code and getting rid of an `else` branch. But if you still object I will change it back.

pedrobaeza replied:
Although I can agree on the principle, this makes more difficult to review the improvements. Maybe you can add an initial commit doing this refactoring, and another with the improvements.

NL66278 committed:
[IMP] *_online_ponto: code organisation

Check situation that should raise exception immediately, prevent ident on normal flow

NL66278 committed:
[FIX] *_online Improvements after review

NL66278 commented:
@pedrobaeza I think I attended to all your comments now.

alexis-via reviewed (approved):
code review, no test

rlizana reviewed (approved):
Code review, @pedrobaeza comments is satisfied, not test!

pedrobaeza commented:
/ocabot merge minor

pedrobaeza commented:
/ocabot merge minor
